src/App/AppZipper.php

```diff
@@ -4,12 +4,40 @@
 
 use RecursiveDirectoryIterator;
 use RecursiveIteratorIterator;
+use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\Filesystem\Path;
 use ZipArchive;
 
 class AppZipper
 {
+    private Filesystem $filesystem;
+
+    public function __construct(private string $cacheDir)
+    {
+        $this->filesystem = new Filesystem();
+    }
+
     public function zip(App $app): string
+    {
+        return $this->cache($app);
```

Comment:
would it make sense to use: https://symfony.com/doc/current/components/cache.html -- for caching?
(use app-hash as key)

Reply:
I kinda thought not because I guess you don't wanna store binary files in different cache backends? file seems to make most sense to me here.

Reply:
I guess serialization in cases like redis could be problematic, yes.  Maybe could then wire in the Filesystem adapter?
assuming it's already available as a dependency in the project.. otherwise, not worth it 👍

Reply:
Sorry, I was just passing by.

IMHO, may we should treat this more like a storage system rather than a cache system 🤔. In the future, we might plan to store all these files in S3 or another service, so using Symfony’s FilesystemAdapter for caching might not be the best fit.

If we want to proceed, could we consider using league/flysystem? like platform did. It supports easy switching between local storage and S3. Or we just keep like in current change and consider improve later.

What do you think?

Reply:
Hey @nam4am - thanks for the input, I think there's not much point to handle this now, because @Gaitholabi is in the midst of building a service registry which will handle all that stuff including most likely s3 storage and in the end we will strip all this stuff out of the SDK to make it as slim as possible.